backend/models/distance.py

```diff
@@ -66,11 +77,105 @@ def get_distance_in_miles(self, point_a, point_b):
         return
 
 
+class OSRMDrivingDistance(MeasureDistance):
+    """
+    Class for OSRM driving distance measurements.
+
+    Uses the `table` OSRM API endpoint.
+    """
+
+    def __init__(self, api_url=None):
+        """Initialize the distance class with the OSRM URL."""
+        self.api_url = os.getenv('OSRM_URL')
+        self._haversine_measurer = HaversineDistance()
+        self.early_exit_outer_radius = 22.0
+
+    @staticmethod
+    def _represent_point_as_str(point):
+        """Represent a point dictionary in the format (longitude,latitude)."""
+        return '{lng},{lat}'.format(lat=point['latitude'], lng=point['longitude'])
+
+    @retry(retry_on_result=_retry_if_result_none, stop_max_attempt_number=10, wait_fixed=2000)
+    def get_distance_in_miles(self, point_a, point_b):
+        """Use an OSRM server to compute the distance and time between two points."""
+        points_as_str = [self._represent_point_as_str(p) for p in (point_a, point_b)]
+        coordinates = ';'.join(points_as_str)
+        url = self.api_url + '/table/v1/car/{coords}'.format(coords=coordinates)
+        response = requests.get(url)
+
+        response.raise_for_status()
+        content = response.json()
+
+        if content['code'] == 'Ok':
```

Comment:
you can simply do  `if content['code'] == 'Ok' and if content['durations']:` instead since you do not have different behaviors for the two exists

Reply:
Done.